DataDog/datadog-lambda-python
# feat: Support urllib3 <= 2.1 for py 3.11

### What does this PR do?
Followup to #329
Fixes #376

### Motivation
Right now boto3 is at 1.26.90 for 3.7, 3.8, 3.9, and 3.10. This causes a known issue when users supply version >= 2.0 of [urllib3](https://urllib3.readthedocs.io/en/latest/v2-migration-guide.html#importerror-cannot-import-name-default-ciphers-from-urllib3-util-ssl).

Previously we pinned urllib3 to all versions of this library, but we'll open this up for Python 3.11.

Also, this marks urllib3 as optional as this library doesn't directly depend on it.

### Testing Guidelines
I tested importing boto3 using urllib3==2.0.7 on python 3.10, it fails to import.
I tested the same function code w/ urllib3==2.0.7 on python 3.11 and the code runes fine.

This pin and the automated tests all pass, so we should be confident this change will open up py3.11 users to urllib3 2.0 without impacting other users using incompatible runtimes (3.7, 3.8, 3.9, and 3.10)

### Additional Notes



### Types of Changes


### Check all that apply
